icu4c/source/common/locid.cpp

```diff
@@ -2459,25 +2459,34 @@ Locale::setKeywordValue(const char* keywordName, const char* keywordValue, UErro
     }
     int32_t bufferLength = uprv_max((int32_t)(uprv_strlen(fullName) + 1), ULOC_FULLNAME_CAPACITY);
     int32_t newLength = uloc_setKeywordValue(keywordName, keywordValue, fullName,
-                                             bufferLength, &status) + 1;
+                                             bufferLength, &status);
     /* Handle the case the current buffer is not enough to hold the new id */
-    if (status == U_BUFFER_OVERFLOW_ERROR) {
-        U_ASSERT(newLength > bufferLength);
-        char* newFullName = (char *)uprv_malloc(newLength);
+    if (status == U_BUFFER_OVERFLOW_ERROR || status == U_STRING_NOT_TERMINATED_WARNING) {
```

Comment:
Hi Frank, I think you missed my big comment from this morning: https://github.com/unicode-org/icu/pull/1461#discussion_r521212124

In short, please change the implementation of uloc_setKeywordValue() back (supposedly) to never setting a not-terminated warning, and then here *assert* that there is no not-terminated warning coming out of that function.

In this function here, other than a new assertion, you can decide whether to just revert it to what it was, or whether to go forward from where you are with cleaning it up a bit. But we should never see a not-terminated warning here and thus need not handle it with special code.

Please see the details in my earlier comment.

Reply:
ok, that make sense. I will work on that and let you know once I repush